core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -769,16 +768,22 @@ object ConsumerGroupCommand extends Logging {
         }
       } else if (opts.options.has(opts.resetToCurrentOpt)) {
         val currentCommittedOffsets = adminClient.listGroupOffsets(groupId)
-        partitionsToReset.map { topicPartition =>
-          currentCommittedOffsets.get(topicPartition).map { offset =>
-            (topicPartition, new OffsetAndMetadata(offset))
-          }.getOrElse(
-            getLogEndOffset(topicPartition) match {
-              case LogOffsetResult.LogOffset(offset) => (topicPartition, new OffsetAndMetadata(offset))
-              case _ => CommandLineUtils.printUsageAndDie(opts.parser, s"Error getting ending offset of topic partition: $topicPartition")
-            }
-          )
+        val (partitionsToResetWithCommittedOffset, partitionsToResetWithoutCommittedOffset) =
+          partitionsToReset.partition(currentCommittedOffsets.keySet.contains(_))
+
+        val preparedOffsetsForParititionsWithCommittedOffset = partitionsToResetWithCommittedOffset.map { topicPartition =>
+          (topicPartition, new OffsetAndMetadata(currentCommittedOffsets.get(topicPartition) match {
+            case Some(offset) => offset
+            case _ => throw new IllegalStateException(s"Expected a valid current offset for topic partition: $topicPartition")
```

Comment:
Currently we print the help message and exit. Is this change intentional?

Reply:
This is another part of the code that should be reachable (because at this point we're certain `currentCommittedOffsets.get(topicPartition)` is not `None`). I used this instead of doing a `currentCommittedOffsets.get(topicPartition).get` above. If you prefer that one instead, or can think of a better way to go about it (that avoids using `.get` on an `Option`) please let me know and I'll make the change. Thanks.

Reply:
Ah, this is for the partitions with committed offsets. This is fine.